ipam/store.go

```diff
@@ -50,3 +51,47 @@ type IpamVm struct {
 	Id uint64 `sql:"AUTO_INCREMENT"`
 	//	IpamHostId sql.NullString
 }
+
+type ipamStore struct {
+	common.DbStore
+}
+
+func (ipamStore *ipamStore) addVm(stride uint, vm *Vm) error {
+	tx := ipamStore.DbStore.Db.Begin()
+
+	row := tx.Model(IpamVm{}).Where("host_id = ? AND segment_id = ?", vm.HostId, vm.SegmentId).Select("IFNULL(MAX(seq),-1)+1").Row()
```

Comment:
Ok, just saw this. So, the use of gorm is really not optional, it's baked into everything we have here. I guess in that case there's no point in trying to abstract away gorm.

We could add a 'query-lib' later on, in which the pre-canned queries we need can be implemented for different backends, I guess. Not important now.